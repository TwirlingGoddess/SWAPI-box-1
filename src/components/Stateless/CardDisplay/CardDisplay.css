.cardCointainer {
  -ms-flex-pack: distribute;
  background-image: url("../../../assets/ea_star_wars_darth_vader.0.jpg");
  background-size: cover;
  display: -ms-flexbox;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  margin: auto;
  min-height: 100vh;
  padding: 11vh; }
