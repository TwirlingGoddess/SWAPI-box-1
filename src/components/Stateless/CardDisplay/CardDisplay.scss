
.cardCointainer {
  padding: 11vh;
  background-image: url('../../../assets/ea_star_wars_darth_vader.0.jpg');
  margin: auto;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-pack: distribute;
  justify-content: space-around;
  background-size: cover;
  flex-wrap: wrap;
  min-height: 100vh;
}
