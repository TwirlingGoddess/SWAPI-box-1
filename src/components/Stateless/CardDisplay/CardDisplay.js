import './CardDisplay.css';
import React from 'react';
import Card from '../Card/Card';
import PropTypes from 'prop-types';
import { Link, Route } from 'react-router-dom';
import CardDetails from '../CardDetails/CardDetails';

const CardDisplay = ({selectedData, findCard, favorites}) => {
  let clicked;
  const displayCards = selectedData.map((data, index) => {
    const findFavorites = favorites.map(favCard => favCard.id);
    if (findFavorites.includes(data.id)) {
      clicked = true;
    } else {
      clicked = false;
    }
    return (
      <Link to={`/starwars/${data.keyList + index}`} key={index}>
        <Card 
          data={data}
          key={index}
          id={data.keyList + index}
          clicked={clicked}
          findCard={findCard}
        />
      </Link>
    );
  });

  return (
    <div className='cardCointainer'>
      <Route path='/starwars/:id' render={({match}) => {
        const { id } = match.params;
        const selectedCard = selectedData.find(card => card.id === id);      
        return (
          <CardDetails {...selectedCard} />
        );
      }} />
      {displayCards}
    </div>
  );
};

CardDisplay.propTypes = {
  selectedData: PropTypes.array,
  findCard: PropTypes.func,
  favorites: PropTypes.array
};

export default CardDisplay;