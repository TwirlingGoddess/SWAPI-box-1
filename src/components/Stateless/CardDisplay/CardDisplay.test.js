import React from 'react';
import { shallow, mount } from 'enzyme';
import CardDisplay from './CardDisplay';

describe('CardDisplay', () => {
  let cardDisplay

  beforeEach(() => {
    cardDisplay = shallow(<CardDisplay />, { disableLifecycleMethods: true });
  })

  it('matches snapshot', () => {
    expect(cardDisplay).toMatchSnapshot()
  })
})
