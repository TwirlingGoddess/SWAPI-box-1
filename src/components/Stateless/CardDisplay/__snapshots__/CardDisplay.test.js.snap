// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDisplay matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardDisplay
    favorites={Array []}
    findCard={[Function]}
    selectedData={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          path="/starwars/:id"
          render={[Function]}
/>,
        Array [],
      ],
      "className": "cardCointainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/starwars/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            path="/starwars/:id"
            render={[Function]}
/>,
          Array [],
        ],
        "className": "cardCointainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/starwars/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
